### PR TITLE
Fix the issues in getting CPU number on IBM i and some build issues o…

### PR DESCRIPTION
**Pull request by bjxyji**
Fix the issues in getting CPU number on IBM i and some build issues on IBM i

Signed-off-by: xiang yang ji <bjxyji@cn.ibm.com>

### CODE REVIEW

**Comment by 0xdaryl** [comment]
Thank you for this contribution.  I think the build issues and the port library changes are independent and should be in separate commits, unless there's a convincing argument otherwise.

File: omrmakefiles/rules.aix.mk [hunk 1]
```diff
@@ -41,6 +41,16 @@ GLOBAL_CFLAGS += -q mbcs -qlanglvl=extended -qarch=ppc -qinfo=pro -qalias=noansi
 GLOBAL_CXXFLAGS+=-q mbcs -qlanglvl=extended -qarch=ppc -qinfo=pro -qalias=noansi -qxflag=LTOL:LTOL0 -qsuppress=1506-1108
 GLOBAL_CPPFLAGS+=-D_XOPEN_SOURCE_EXTENDED=1 -D_ALL_SOURCE -DRS6000 -DAIXPPC -D_LARGE_FILES
 
+ifdef I5_VERSION
+  I5_FLAGS+=-g -qtbtable=full -qlist -qsource 
+  I5_DEFINES+=-DJ9OS_I5 -DJ9OS_$(I5_VERSION) -I$(top_srcdir)/../iseries -I$(top_srcdir)/../oti
```
Comment (charliegracie) [inline omrmakefiles/rules.aix.mk]: This line looks J9 Java specific.  If that is the case the code should be made in the J9 Java language extensions.

File: tools/hookgen/makefile [hunk 2]
```diff
@@ -30,4 +30,11 @@ MODULE_INCLUDES += $(OMR_PUGIXML_DIR)
 vpath %.cpp $(OMR_PUGIXML_DIR)
 vpath %.c $(OMR_PUGIXML_DIR)
 
+# HookGen build failed with pase vac13 and it only run on AIX.
+# So, we use the default vac on AIX build system
+ifdef I5_VERSION
```
Comment (charliegracie) [inline tools/hookgen/makefile]: I am not sure that it is best to change the compiler here.  Can you provide more details what is wrong with the default compiler?

File: port/unix/omrsysinfo.c [hunk 3]
```diff
@@ -1099,13 +1099,20 @@ omrsysinfo_get_number_CPUs_by_type(struct OMRPortLibrary *portLibrary, uintptr_t
 		break;
 
 	case OMRPORT_CPU_TARGET: {
+#if defined(J9OS_I5)
+		toReturn = _system_configuration.ncpus;
```
Comment (charliegracie) [inline port/unix/omrsysinfo.c]: As per @0xdaryl comment could this change be pulled out into a separate commit.  It is fine to be in the same Pull Request
